modules/sdk/project-templates/project-templates/src/test/java/com/liferay/project/templates/ProjectTemplatesTest.java

```diff
@@ -1003,6 +1005,132 @@ public void testBuildTemplateServiceBuilderNestedPath() throws Exception {
 			":modules:nested:path:sample");
 	}
 
+	@Test
+	public void testBuildTemplateServiceBuilderVersions() throws Exception {
+		String projectPath = ":modules:nested:path:sample";
+
+		String name = "sample";
+
+		String packageName = "com.test.sample";
+
+		File workspaceProjectDir = _buildTemplateWithGradle(
+			WorkspaceUtil.WORKSPACE, "ws-nested-path");
+
+		File destinationDir = new File(
+			workspaceProjectDir, "modules/nested/path");
+
+		Assert.assertTrue(destinationDir.mkdirs());
+
+		File gradleProjectDir = _buildTemplateWithGradle(
+			destinationDir, "service-builder", name, "--package-name",
+			packageName);
+
+		_testContains(
+			gradleProjectDir, "sample-service/build.gradle",
+			"compileOnly project(\":modules:nested:path:sample:sample-api\")");
+
+		final String serviceProjectName = name + "-service";
+
+		String[] gradleSericeBuilderVersion = new String[1];
+
+		_testChangePortletModelHintsXml(
+			gradleProjectDir, serviceProjectName,
+			new Callable<Void>() {
+
+				@Override
+				public Void call() throws Exception {
+					_executeGradle(
+						workspaceProjectDir,
+						projectPath + ":" + serviceProjectName +
+							_GRADLE_TASK_PATH_BUILD_SERVICE);
+
+					String serviceBuilderDep = null;
+
+					String result = _executeGradle(
+						workspaceProjectDir,
+						projectPath + ":" + serviceProjectName +
+							":dependencies");
+
+					Scanner scanner = new Scanner(result);
```

Comment:
Use a Regex pattern instead of a scanner.  see here for example: https://github.com/Ithildir/liferay-portal/pull/1023/commits/0a30d53bbb253a9761d764906f257dee10bd0f82